### PR TITLE
Add support for 'pointerlockerror' event

### PR DESCRIPTION
**Pull request by rfeneon**
This was missing from the HTML5 Pointerlock API.

Without this event the application has no way to know if the pointer lock request has failed.

There is a new test: "interactive.test_html5_pointerlockerror".

P.-S.: I have noticed that there are some tracked files in site/build/text/docs/, is this deliberate?

### CODE REVIEW

File: tests/test_html5_pointerlockerror.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  printf("SUCCESS! received 'pointerlockerror' event\n");
+  report_result(0);
+
+  return 0;
+}
+
+int main()
+{
+  printf("'pointerlockerror' event test:\n");
+  printf("Disable mouse pointer lock for this page then click on canvas\n");
```
Comment (juj) [inline tests/test_html5_pointerlockerror.c]: Hmm, how does one disable mouse pointer lock for a page (before clicking)? Or does this mean: first click on a canvas and then reject the pointer lock request?

**Comment by juj** [comment]
Thanks, this looks good. One minor question above about the test message?

File: tests/test_html5_pointerlockerror.c (hunk 1 above)
Comment (rfeneon) [inline tests/test_html5_pointerlockerror.c]: You can do that on Chrome by clicking the connection icon in the address bar, but I didn't check if it's possible with the other browsers.
Maybe I should clarify the test message, something like;
"Disable mouse pointer lock for this page then click on canvas, or reject the pointer lock request after clicking on canvas" ?

Comment (juj) [inline tests/test_html5_pointerlockerror.c]: Yeah, that sounds good. I think user might be able to reject pointer lock only after the site has requested it first.

**Comment by rfeneon** [comment]
Updated test message.

**Comment by juj** [comment]
thanks!

**Comment by juj** [comment]
Oh, just noticed your other question. Not sure if there is any special reason to have those. Submitted https://github.com/kripken/emscripten/pull/4409 to clean it up.

**Comment by rfeneon** [comment]
thanks for your review!

**Pull request merged**